<!-- poseslam testing-->
<launch>

<!--POSE SLAM-->
<node name="iri_poseslam"
      pkg="iri_poseslam"
      type="iri_poseslam"
      output="screen">

      <param name="~loop_rate"  value="20.0"/>
      
      <param name="~match_area_x"  value="2.0"/> <!--3.0-->
      <param name="~match_area_y"  value="2.0"/> <!--3.0-->
      <param name="~match_area_th" value="0.3"/> <!--0.261799388-->

      <param name="~pd_range_1" value="0.2"/> <!--usually 0.2 - If the probability of 2 poses of being closer than 'matchArea' is larger than 'pd.first' we try to create a loop.-->
      <param name="~pd_range_2" value="0.9"/> <!--usually 0.9 - If the probability of 2 poses of being closer than 'matchArea' is larger than 'pd.second' they are close enough for one of them to be redundant-->
      <param name="~ig_range_1" value="2.0"/> <!--usually 7.0 - If the information gain is smaller than 'ig.first' the pose is redundant and it can be overwrite next time slice-->
      <param name="~ig_range_2" value="5.0"/> <!--usually 11.0 - If the information gain is larger than 'ig.second' we try to create a loop-->

      <param name="~closing_loop_noise_xx"   value="7.25483e-06"/>
      <param name="~closing_loop_noise_xy"   value="-8.98881e-07"/>
      <param name="~closing_loop_noise_xth"  value="5.23212e-07"/>
      <param name="~closing_loop_noise_yy"   value="9.29614e-06"/>
      <param name="~closing_loop_noise_yth"  value="-1.09644e-06"/>
      <param name="~closing_loop_noise_thth" value="4.70196e-07"/>

      <param name="~ignore_previous_steps"   value="4"/>

      <param name="~initial_position_x"    value="10.7356"/>
      <param name="~initial_position_y"    value="2.188"/>

      <param name="~results_path"          value="/home/jvallve/Dropbox/Juan\ Andrade\ -\ Joan\ VallvÃ©/ICRA14Rafael/resultadosROS/static_results.m"/>



      <remap from="~get_link" to="/sensors_2_link/get_link"/>

</node>

<!--SERVICES AND PUBLISHERS-->

<node name="odom_2_odomrel"
      pkg="iri_poseslam"
      type="odom_2_odomrel"
      output="screen">
      <remap from="~odom" to="/vmc_navserver/odom" />

      <param name="~covariance_factor"  value="0.01"/>
</node>

<node name="sensors_2_link"
      pkg="iri_poseslam"
      type="sensors_2_link"
      output="screen">
      <remap from="~scan" to="/laserscan" /> 
      <remap from="~odom_relative" to="/odom_2_odomrel/odom_rel" />
      <remap from="~get_relative_pose" to="/iri_laser_icp/get_relative_pose" />

      <param name="~online_mode"                      value="false"/>
      <param name="~fusion_mode"                      value="3"/>
      <param name="~N_scans_discard"                  value="5"/>
      <param name="~ICP_covariance_correction_factor" value="100.0"/>
      <param name="~cmd_vel_available"                value="false"/>
      <param name="~dx_base_2_laser"                  value="0.695"/><!-- 0.695-->
      <param name="~dy_base_2_laser"                  value="-0.01"/><!-- -0.01-->
      <param name="~dth_base_2_laser"                 value="-0.0069813"/><!-- -0.0069813-->
      <param name="~dz_base_2_laser"                  value="0"/>
      <param name="~dz_footprint_2_base"              value="0"/>
</node>

<node name="iri_laser_icp"
      pkg="iri_laser_icp"
      type="iri_laser_icp"
      output="screen">
    <param name="~max_angular_correction_deg" value="50"/>
    <param name="~max_linear_correction"      value="3.0"/>
    <param name="~max_iterations"             value="50"/>
    <param name="~max_correspondence_dist"    value="3.0"/>
    <param name="~clustering_threshold"       value="0.1"/>
    <param name="~orientation_neighbourhood"  value="200"/>
    <param name="~use_point_to_line_distance" value="0"/>
    <param name="~use_corr_tricks"            value="1"/>
    <param name="~restart"                    value="1"/>
    <param name="~do_visibility_test"         value="0"/>
    <param name="~do_compute_covariance"      value="1"/>
    <param name="~epsilon_xy"                 value="0.000000001"/>
    <param name="~epsilon_theta"              value="0.000000001"/>
    <param name="~debug_verify_tricks"        value="0"/>
    <param name="~outliers_adaptive_order"    value="0.8"/>
    <param name="~min_laser_range"            value="1.0"/>
    <param name="~max_laser_range"            value="1.0e10"/>
</node>


<!--RESULTS VISUALIZATION-->
<node name="rviz"
      pkg="rviz"
      type="rviz"
      args="-d $(find iri_poseslam)/config/poseslam_RAFA.vcg" />

<node name="trajectory_scans_2_pointcloud"
      pkg="iri_poseslam"
      type="trajectory_scans_2_pointcloud"
      output="screen">
      <remap from="~scan" to="/laserscan" /> 
      <remap from="~trajectory" to="/iri_poseslam/trajectory" />
      <param name="~publish_redundant"  value="true"/>
</node>

<node name="trajectory_2_markers"
      pkg="iri_poseslam"
      type="trajectory_2_markers"
      output="screen">
      <remap from="~trajectory" to="/iri_poseslam/trajectory" />
      <param name="~dz_base_2_laser"                  value="0"/>
      <param name="~dz_footprint_2_base"              value="0"/>
</node>

<node pkg="tf" type="static_transform_publisher" name="world2map" args="0 0 0 0 0 0 /world /map 100" />

<!--SIMULATION-->

<param name="use_sim_time" value="true" />

<node pkg="rosbag"
      type="play"
      name="rosbag"
      args="-d 10 --clock -k -r 1 $(find iri_poseslam)/data/boxing_new.bag"/>
      <!--
      static1_2013-09-05-15-25-31
      dynamic1_2013-09-04-17-08-57
      boxing_new
      -->
</launch>
